### PR TITLE
Bugfix: Drain machines with only a valid NodeName

### LINKED ISSUE #465: MCM should not try to drain machines that have not joined the cluster
**What happened**:
MCM is trying to delete some `Machine` objects. Now it tries to drain the corresponding `Node` objects, however, as the mentioned `Machine`s never joined the cluster, there are no such `Node` objects. The drain fails forever with:

```
status:
  currentStatus:
    lastUpdateTime: "2020-05-28T15:30:18Z"
    phase: Terminating
  lastOperation:
    description: Drain failed - resource name may not be empty
    lastUpdateTime: "2020-05-28T15:45:52Z"
    state: Failed
    type: Delete
```

Logs of the machine-controller-manager:

```
I0528 15:45:42.432992       1 deployment.go:448] Processing the machinedeployment "shoot--foo--bar-cpu-worker" (with replicas 4)
W0528 15:45:42.639551       1 machine.go:658] Drain failed for machine "shoot--foo--bar-cpu-worker-7cdf986ff9-pzcgs".
Buf:
ErrBuf:
Err-Message:resource name may not be empty
W0528 15:45:42.785280       1 machine.go:658] Drain failed for machine "shoot--foo--bar-cpu-worker-7cdf986ff9-62mcz".
Buf:
ErrBuf:
Err-Message:resource name may not be empty
W0528 15:45:42.836888       1 machine.go:658] Drain failed for machine "shoot--foo--bar-cpu-worker-7cdf986ff9-k46s5".
Buf:
ErrBuf:
Err-Message:resource name may not be empty
I0528 15:45:42.883235       1 machine.go:551] Deleting Machine "shoot--foo--bar-cpu-worker-7cdf986ff9-pzcgs"
E0528 15:45:42.883357       1 drain.go:193] Error getting details for node: "". Err: resource name may not be empty
I0528 15:45:42.883371       1 drain.go:175] Machine drain ended on 2020-05-28 15:45:42.883368449 +0000 UTC m=+47.035130779 and took 90.578µs for ""
```

**What you expected to happen**:
MCM should not try to drain these nodes.

**How to reproduce it (as minimally and precisely as possible)**:
Create machine objects that won't join the cluster and then try to delete them.

**Environment**:
MCM v0.29.0

### PR DESCRIPTION
**Pull request by prashanth26**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #465 

**Special notes for your reviewer**:

**Release note**:

```improvement developer
Bugfix: Drain machines with only a valid node (name)
```

### CODE REVIEW

File: pkg/controller/machine.go [hunk 1]
```diff
@@ -599,7 +599,9 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.Driv
 			}
 		}
 
-		if machineID != "" {
+		if machineID != "" && nodeName != "" {
+			// Begin drain logic only when the nodeName & providerID exist's for the machine
```
Comment (hardikdr) [inline pkg/controller/machine.go:603]:
```suggestion
			// Begin drain logic only when the nodeName & providerID exists for the machine
```

Comment (hardikdr) [reply pkg/controller/machine.go:603]: Do we need this change also in OOT?

**Comment by prashanth26** [comment]
> Do we need this change also in OOT?

Nice catch. Even I thought about it, although I think the OOT will not run into such a situation. I have added this check just to be safe. Here - https://github.com/gardener/machine-controller-manager/pull/480/commits/66fee5c42d97e3d938dd744150eb05b46a6726d4

Also if all looks good, please squash and merge it. Thanks.

**Comment by prashanth26** [comment]
Hi @ggaurav10 ,

Nice catch with this. Made the suggested change here - https://github.com/gardener/machine-controller-manager/pull/480/commits/0a7d3b5cd34c19ebcce973d1f5307a372332c8d5

**Pull request merged**